src/dune_pkg/solver_env.ml

```diff
@@ -86,3 +116,48 @@ let to_env t variable =
     in
     get t variable_name |> Option.map ~f:Variable_value.to_opam_variable_contents
 ;;
+
+let popular_platform_envs =
+  let make ~os ~arch ?os_distribution ?os_family () =
```

Comment:
Could you please make these arguments required? I don't think you're saving yourself a lot of typing here, and I would like to see where we pass `os` and `arch` but omit `os_distribution` and `os_family`.